keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -92,8 +132,7 @@ class ArtifactListener(
 
       if (latestArtifact != null) {
         log.debug("Storing latest version {} (status={}) for registered artifact {}", latestArtifact.version, latestArtifact.status, artifact)
-        val enrichedArtifact = artifactSupplier.addMetadata(latestArtifact.normalized())
-        repository.storeArtifactVersion(enrichedArtifact)
+        enrichAndStore(latestArtifact.normalized(), artifactSupplier)
```

Comment:
Looks like you're always calling `normalized()` before calling `enrichAndStore()` -- you could probably just move the call to the former there.

Reply:
Good call